src/connector/src/parser/unified/json.rs

```diff
@@ -446,6 +446,17 @@ impl JsonParseOptions {
                     .collect::<Result<_, _>>()?,
             )
             .into(),
+
+            // String containing json object, e.g. "{\"a\": 1, \"b\": 2}"
+            // Try to parse it as json object.
+            (Some(DataType::Struct(_)), ValueType::String) => {
+                // TODO: avoid copy by accepting `&mut BorrowedValue` in `parse` method.
+                let mut value = value.as_str().unwrap().as_bytes().to_vec();
+                let value =
+                    simd_json::to_borrowed_value(&mut value[..]).map_err(|_| create_error())?;
+                return self.parse(&value, type_expected);
+            }
+
```

Comment:
In the description, the schema should be declared as `nested struct <>` but there is no option for `nested`

Reply:
Ahh, `nested` is the name of the column as an example. 😂